Microsoft.SystemForCrossDomainIdentityManagement/Protocol/ProtocolExtensions.cs

```diff
@@ -102,90 +103,192 @@ private static void Apply(this Core2Group group, PatchOperation2 operation)
+            {
+                group.DisplayName = null;
+            }
+            else
+            {
+                group.DisplayName = value.Value;
+            }
+        }
+
+        private static void MemberPatchOperation(this Core2Group group, PatchOperation2 operation)
+        {
+            if (operation.Value != null)
+            {
+                switch (operation.Name)
+                {
+                    case OperationName.Replace:
+                        group.ReplaceMembersOperation(operation);
+                        break;
+
+                    case OperationName.Add:
+                        group.AddMembersOperation(operation);
+                        break;
+
+                    case OperationName.Remove:
+                        group.RemoveMembersOperation(operation);
+                        break;
+                }
+            }
+        }
+
+        private static void AddMembersOperation(this Core2Group group, PatchOperation2 operation)
+        {
+            IEnumerable<Member> newMembers = operation.Value
+                 .Select(
+                     (OperationValue item) =>
+                         new Member()
+                         {
+                             Value = item.Value
+                         })
+                 .ToArray();
+
+            if (group.Members != null)
+            {
+                //merge existing with new members
+                var existingMembers = group.Members.Distinct(new MemberComparer()).ToDictionary(k => k.Value, v => v);
+                foreach (var newMember in newMembers)
+                {
+                    existingMembers[newMember.Value] = newMember;
+                }
+                group.Members = existingMembers.Values;
+            }
+            else
+            {
+                //no existing members, so just use new members.
+                group.Members = newMembers;
+            }
+        }
+        private static void ReplaceMembersOperation(this Core2Group group, PatchOperation2 operation)
+        {
+            IEnumerable<Member> replacementMembers =
+                 operation
+                 .Value
+                 .Select(
+                     (OperationValue item) =>
+                         new Member()
+                         {
+                             Value = item.Value
+                         })
+                 .ToArray();
+
+            //replace - so just use new members.
+            group.Members = replacementMembers;
+        }
+
+        private static void RemoveMembersOperation(this Core2Group group, PatchOperation2 operation)
+        {
+            if (group.Members == null)
+            {
+                return;
+            }
+
+            //patch is asking to remove members completely.
+            if (operation.Value?.FirstOrDefault()?.Value == null &&
+                (operation.Path?.SubAttributes == null || operation.Path?.SubAttributes.Count == 0))
```

Comment:
Instead of the extensive logic here, the deserialization of the incoming request into PatchOperation2 object needs to be changed so that it can handle the singled valued removal that you mentioned in the description of this pull request. We'll make changes to handle that soon.